pyproject.toml

```diff
@@ -47,7 +47,7 @@ dask = { version = ">=2021.10.0", optional = true, extras = [
   "array",
   "dataframe"
 ] }
-datafusion = { version = ">=0.4,<0.7", optional = true }
+datafusion = { version = ">=0.6,<0.7", optional = true }
```

Comment:
Since the lockfile didn't change, that means we're testing with 0.6 so I don't think this will help the issue. Can you just xfail only on versions older than 0.6?